### PR TITLE
rpc: prevent spending unconfirmed funds within OpenChannel by default

### PR DESCRIPTION
**Pull request by wpaulino**
In this commit, we address a slight regression in the defaults provided
by the OpenChannel RPC with regards to spending unconfirmed outputs to
fund funding transactions. We now add a new boolean parameter to
OpenChannelRequest: SpendUnconfirmed. If set, we'll use this to
indicate to the server that the funding transaction of the channel to be
created can spend the wallet's unconfirmed outputs to fund it. This
addition is needed because otherwise, if the caller doesn't specify a
MinConfs parameter to the request, then a default of 0 will be used,
which is not ideal.

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -761,6 +761,22 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
 			"non-negative number")
 	}
 
+	var minConfs int32
```
Comment (halseth) [inline rpcserver.go]: Doesn't look like it is ever set in the case `in.MinConfs > 0`

Comment (wpaulino) [reply rpcserver.go]: Fixed!

File: rpcserver.go [hunk 2]
```diff
@@ -928,10 +943,25 @@ func (r *rpcServer) OpenChannelSync(ctx context.Context,
 			"size is: %v SAT", int64(minChanFundingSize))
 	}
 
+	minConfs := in.MinConfs
```
[the remark this answers is not in the document]
Comment (wpaulino) [reply rpcserver.go]: Fixed.

File: rpcserver.go [hunk 3]
```diff
@@ -755,10 +755,25 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
...
+	case minConfs < 0:
 		return errors.New("minimum number of confirmations must be a " +
 			"non-negative number")
+
+	// The funding transaction should not be funded with unconfirmed outputs
+	// unless explicitly specified by SpendUnconfirmed. We do this to
+	// provide sane defaults to the OpenChannel RPC, as otherwise, if the
+	// MinConfs field isn't explicitly set by the caller, we'll use
+	// unconfirmed outputs without the caller being aware.
+	case in.MinConfs == 0 && !in.SpendUnconfirmed:
```
[the remark this answers is not in the document]
Comment (wpaulino) [reply rpcserver.go]: Fixed, we'll now return an error to the caller indicating so.

**Pull request merged**